### PR TITLE
Debugging SQL Server in Actions

### PR DESCRIPTION
**Pull request by paulbalandan**
**Checklist:**
- [x] Securely signed commits

### CODE REVIEW

**Comment by MGatner** [comment]
This one is a little hard to check on mobile, I will take a look later on desktop. Is the Action change intentional? Or is that part of the other PR?

**Comment by paulbalandan** [comment]
> This one is a little hard to check on mobile, I will take a look later on desktop. Is the Action change intentional? Or is that part of the other PR?

I'm planning to consolidate the two PRs as their goal is to determine where went wrong.

File: app/Views/errors/html/error_exception.php [hunk 1]
```diff
@@ -19,7 +19,7 @@
 	<!-- Header -->
 	<div class="header">
 		<div class="container">
-			<h1><?= esc($title), esc($exception->getCode() ? ' #' . $exception->getCode() : '') ?></h1>
+			<h1><?= esc($title) ?></h1>
```
Comment (kenjis) [inline app/Views/errors/html/error_exception.php:22]: Why do you remove the exception code?

Comment (paulbalandan) [reply app/Views/errors/html/error_exception.php:22]: This was made to make the output similar to its CLI equivalent.

Comment (paulbalandan) [reply app/Views/errors/html/error_exception.php:22]: Should have been a separate PR, I guess?

**Comment by paulbalandan** [comment]
This one is very tricky to debug. It seems PHP < 8.0 prefers 2017 image while PHP 8+ prefers 2019 image. 🤯

**Comment by MGatner** [comment]
I'm very grateful you are hammering on this one. Sorry it has been so stubborn 😳 I have very little MSSQL experience but let me know if there's some way I can support you.

**Comment by paulbalandan** [comment]
If there's someone who can imitate the setup locally on a Docker container running on Ubuntu 20.04 and pulling the latest 2019 MSSQL image from MCR and check how to connect our PHP testing to the container. I have tried every possible combination of solutions as detailed in the Microsoft docs and everywhere else and it seems this has been very stubborn. Any help or direction is appreciated.

**Comment by MGatner** [comment]
I don't know if @michalsn is around, but if you are and have a moment to take a look I know you have some expertise in this area.

File: .github/workflows/test-phpunit.yml [hunk 2]
```diff
@@ -62,13 +62,14 @@ jobs:
...
-        image: microsoft/mssql-server-linux:2017-latest
+        image: mcr.microsoft.com/mssql/server:2019-CU10-ubuntu-20.04
         env:
-          SA_PASSWORD: 1Secure*Password1
+          MSSQL_SA_PASSWORD: 1Secure*Password1
           ACCEPT_EULA: Y
           MSSQL_PID: Developer
         ports:
           - 1433:1433
+        options: --name=mssql --health-cmd="/opt/mssql-tools/bin/sqlcmd -S 127.0.0.1,1433 -U sa -P '1Secure*Password1' -Q 'SELECT @@VERSION'" --health-interval=10s --health-timeout=5s --health-retries=3
```
Comment (michalsn) [inline .github/workflows/test-phpunit.yml:72]:
```suggestion
        options: --health-cmd="/opt/mssql-tools/bin/sqlcmd -S 127.0.0.1 -U sa -P 1Secure*Password1 -Q 'SELECT @@VERSION'" --health-interval=10s --health-timeout=5s --health-retries=3
```

File: .github/workflows/test-phpunit.yml [hunk 3]
```diff
@@ -92,7 +93,7 @@ jobs:
         with:
           php-version: ${{ matrix.php-versions }}
           tools: composer, pecl
-          extensions: imagick, sqlsrv-beta, gd, sqlite3, redis, memcached, pgsql
+          extensions: imagick, sqlsrv-5.9.0preview1, gd, sqlite3, redis, memcached, pgsql
```
Comment (michalsn) [inline .github/workflows/test-phpunit.yml:96]:
```suggestion
          extensions: imagick, sqlsrv, gd, sqlite3, redis, memcached, pgsql
```

**Comment by michalsn** [comment]
Seems to work ok: https://github.com/michalsn/CodeIgniter4/commit/2bcb82885292804be4160612fb30c4a733159df3
https://github.com/michalsn/CodeIgniter4/actions/runs/774899370

**Comment by paulbalandan** [comment]
Thanks, @michalsn ! Seems all resolved now.

**Review by samsonasik — APPROVED** [review]

**Pull request merged**